### PR TITLE
Fix terraform apply

### PR DESCRIPTION
**Pull request by dekiel**
**Description**

### <samp>🤖 Generated by Copilot at a77d7d3</samp>

The pull request refactors and simplifies the Terraform-related tasks and pipelines in the `test-infra` repository. It introduces a new parameter for passing custom arguments to the `tfcmt` tool, which comments on pull requests with Terraform plan output. It also renames the `hide-github-comments` task and step to `hide-github-pr-comments` to clarify its functionality.

Changes proposed in this pull request:

### <samp>🤖 Generated by Copilot at a77d7d3</samp>

*  Rename `hide-github-comments` task and step to `hide-github-pr-comments` to clarify scope ([link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-9c8e66e1107849af36a0453366910cda7c5505c6e42b9232f164fd0d5cf91256L1-R3), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-b2bcfeb88d1bf36e5babb0ab7e9be431860cfe5ac3007b52090b33539457e674L4-R4), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-b2bcfeb88d1bf36e5babb0ab7e9be431860cfe5ac3007b52090b33539457e674L10-R16), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-b2bcfeb88d1bf36e5babb0ab7e9be431860cfe5ac3007b52090b33539457e674L35-R35), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-b6ef9ae7e43d1372472e9ce64e29a15131f2fb77eea3ba34be616040bfed0184L63-R69), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-b6ef9ae7e43d1372472e9ce64e29a15131f2fb77eea3ba34be616040bfed0184L147-R149), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-068dcee5d265badf56c0be3924d1871451b64916699f6c52fc85cdee51928d42L63-R69), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-068dcee5d265badf56c0be3924d1871451b64916699f6c52fc85cdee51928d42L147-R149))
* Remove `hide-github-pr-comments` step from `postsubmit` pipeline in `prow/jobs/test-infra/terraform.yaml` and `task/terraform-executor/samples/sample_prowjob_pipeline.yaml`, as it is not needed for merged pull requests ([link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-256311d5c90525ccdc59820d6b49afc1f8d38b00c125518733eef9da923a6941L149-R151), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-b6ef9ae7e43d1372472e9ce64e29a15131f2fb77eea3ba34be616040bfed0184L147-R149), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-068dcee5d265badf56c0be3924d1871451b64916699f6c52fc85cdee51928d42L147-R149))
* Remove `SHA` and `PULL_NUMBER` parameters from `terraform-executor` task and step, and replace them with `additional_tfcmt_args` parameter to allow passing arbitrary arguments to `tfcmt` tool ([link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-465ad63da19892653ec7291209ccd207b69b7e7cb99345cf4d19ba42ce35e2fbL30-R31), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-df5e9330e4489520fa2fc2bed682de1c2322b830944aee99d51b84d5d77a05d3R19-R20), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-df5e9330e4489520fa2fc2bed682de1c2322b830944aee99d51b84d5d77a05d3L36-R40), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-df5e9330e4489520fa2fc2bed682de1c2322b830944aee99d51b84d5d77a05d3L74-R73), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-df5e9330e4489520fa2fc2bed682de1c2322b830944aee99d51b84d5d77a05d3L82-R80), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-256311d5c90525ccdc59820d6b49afc1f8d38b00c125518733eef9da923a6941L64-R70), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-256311d5c90525ccdc59820d6b49afc1f8d38b00c125518733eef9da923a6941L149-R151), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-b6ef9ae7e43d1372472e9ce64e29a15131f2fb77eea3ba34be616040bfed0184L63-R69), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-b6ef9ae7e43d1372472e9ce64e29a15131f2fb77eea3ba34be616040bfed0184L147-R149), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-068dcee5d265badf56c0be3924d1871451b64916699f6c52fc85cdee51928d42L63-R69), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-068dcee5d265badf56c0be3924d1871451b64916699f6c52fc85cdee51928d42L147-R149))
* Update `description` and `tags` fields of `terraform-executor` task and `hide-github-pr-comments` task to reflect changes and usage ([link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-b2bcfeb88d1bf36e5babb0ab7e9be431860cfe5ac3007b52090b33539457e674L4-R4), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-b2bcfeb88d1bf36e5babb0ab7e9be431860cfe5ac3007b52090b33539457e674L10-R16), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-465ad63da19892653ec7291209ccd207b69b7e7cb99345cf4d19ba42ce35e2fbL30-R31), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-df5e9330e4489520fa2fc2bed682de1c2322b830944aee99d51b84d5d77a05d3R19-R20))
* Update installation command and sample pipeline in `README.md` files of `terraform-executor` task and `hide-github-pr-comments` task to use new names and parameters ([link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-9c8e66e1107849af36a0453366910cda7c5505c6e42b9232f164fd0d5cf91256L13-R13), [link](https://github.com/kyma-project/test-infra/pull/7755/files?diff=unified&w=0#diff-465ad63da19892653ec7291209ccd207b69b7e7cb99345cf4d19ba42ce35e2fbL30-R31))

/kind bug
/area automation

### CODE REVIEW

**Comment by dekiel** [comment]
This pr fixes terraform-executor tekton tasks.
/override pull-plan-prod-terraform

**Review by Sawthis — APPROVED** [review]

File: task/terraform-executor/0.1/README.md [hunk 1]
```diff
@@ -27,9 +27,8 @@ kubectl apply -f https://raw.githubusercontent.com/kyma-project/test-infra/main/
 - **terraform_action**: Terraform action to execute on provided config files. _(REQUIRED)_
 - **module_path**: Path to the Terraform config files. _(REQUIRED)_
 - **additional_terraform_args**: Additional Terraform arguments. Add `-no-color` flag if you override the default value. _(OPTIONAL, default: '[ "-no-color" ]')_
+- **additional_tfcmt_args**: Additional tfcmt arguments. Allows defining different set of arguments depending on the use case. Parameter must provide at least -sha and/or -pr arguments with values._(REQUIRED)_
```
Comment (IwonaLanger) [inline task/terraform-executor/0.1/README.md:30]:
```suggestion
- **additional_tfcmt_args**: Additional tfcmt arguments. Allows defining a different set of arguments depending on the use case. The parameter must provide at least `-sha` and/or `-pr` arguments with values._(REQUIRED)_
```

File: task/terraform-executor/0.1/terraform-executor.yaml [hunk 2]
```diff
@@ -16,6 +16,8 @@ spec:
     The usual actions are `plan` and `apply`. Authentication to Google Cloud is done through the Kubernetes service account with the
     workload identity configured. The parameter **additional_terraform_args** allows defining Terraform CLI arguments in pipelines.
     When this parameter value is set in a pipeline, you must add the default value `-no-color` to the list.
+    The parameter **additional_tfcmt_args** allows defining additional variable tfcmt CLI arguments in pipelines.
+    This parameter must at least provide -sha and/or -pr arguments together with its values. See example pipeline for parameters values example.
```
Comment (IwonaLanger) [inline task/terraform-executor/0.1/terraform-executor.yaml:20]: I'm not sure I understand this line correctly, but I think that, depending on the number of parameters in question, either ` This parameter must at least provide -sha and/or -pr arguments together with its values. See the example pipeline for the parameter's values example.`, or ` This parameter must at least provide -sha and/or -pr arguments together with its values. See the example pipeline for the parameters' values example` should sound better.

**Review by Sawthis — APPROVED** [review]

**Review by IwonaLanger — APPROVED** [review]

**Pull request merged**